Codecademy/gamut
# fix: maintain open state when hovering over 'floating' tooltip content

### Overview


Adds event listeners to mouse overing a floating tooltip so it remains open.
Previously, the floating variant would close whenever mousing over the tooltip content.


### PR Checklist

- [x] Related to JIRA ticket: [DOT-79]

#### Testing Instructions



1. visit the [prod storybook page](https://gamut.codecademy.com/?path=/docs/atoms-tooltip--tool-tip) for the tooltip.  Go to the first example and set the `placement` as `floating`.  Note that mousing over the tooltip example causes it to close (most of the time).
![Screenshot 2024-02-29 at 4 13 06 PM](https://github.com/Codecademy/gamut/assets/34633964/28373f31-c172-4f09-9927-ff79e91b8d46)


2. now visit the [new storybook page](https://65e102132724213c07d8b869--gamut-preview.netlify.app/?path=/docs/atoms-tooltip--tool-tip) and confirm the fix - tooltip should stay open when mousing over the popover content.
![new_storybook](https://github.com/Codecademy/gamut/assets/34633964/bd94fc85-489b-48eb-95f9-c74c2964257c)


#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37212)  | [Monolith Env](https://pr-37212-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/5216)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-5216)   |




[DOT-79]: https://codecademy.atlassian.net/browse/DOT-79?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
